parser/model/ddl.go

```diff
@@ -902,6 +902,9 @@ type SchemaDiff struct {
 	RegenerateSchemaMap bool `json:"regenerate_schema_map"`
 
 	AffectedOpts []*AffectedOption `json:"affected_options"`
+
+	// The start ts of the transaction that writes the SchemaDiff, used as the timestamp that the schema starts to take effect
```

Comment:
Not exactly. The timestamp that actually "starts to take effect", is the commitTS corresponding to this `startTS`. Try to query `get_mvcc_by_ts/{starTS}/{m.schemaDiffKey(schemaVersion)}`, see if you can get the desired information.

Reply:
> > thanks! i updated to store startTs in schemadiff and use it as the timestamp of a schema version, i haven't tested it yet but want to get your opinion first to see if this is what you expect?
> 
> You still need to use `commitTS`. Otherwise LGTM.

try to get some clarity, do you suggest to use https://github.com/pingcap/tidb/pull/42083/commits/c169d462e80c104b7965bcb8c30e60beeb82234f ? that version uses the commit ts of the schema diff, and don't need to handle backward compatibility because the commit ts is already available

Reply:
OK, OK.

I think I overlooked the commit message. I thought you are still using startTS to retrive timestamp.

Yes, that commit looks fine. I'll take a deeper look tomorrow.

Reply:
@hihihuhu OK, sorry. :(
I did overlooked the commit. That commit is fine. Please change to that commit.